docs/adr/0019-use-1password-for-sharing-secrets.md

```diff
@@ -0,0 +1,96 @@
+  control, and sometimes without warning. This creates a challenge: how can we
+  share an updated secret in a secure way with minimal interruption to ongoing
+  app engineering work?
+
+* Decision drivers:
+  1. EASi engineers need to be able to stay up-to-date on secret values that
+     are required for app development.
+  1. There should be a central source of truth for the secrets.
+  1. The scheme for storing, sharing, and retrieving secrets should be
+     consistent with our understanding of best security practices.
+  1. The scheme should also be user-friendly and accessible without assuming
+     any knowledge of back-end infrastructure or overly complex authentication
+     mechanisms.
+
+## Considered Alternatives
+
+* Email
+* Slack DMs
+* 1Password
+* AWS SSM / ParameterStore
+
+## Decision Outcome
+
+Chosen Alternative: Use a 1Password vault for sharing secrets.
+
+* 1Password allows us to control access to the secrets repository on a per-user
+  basis, and updates to any secret will be available to other users on the
+  vault immediately.  With a separate vault within 1Password, we need not share
+  secrets with other Truss personnel who are not assigned to the project.
+* 1Password also has familiar and user-friendly graphical interfaces via the
+  web and a multi-platform desktop application.
+* Access to secrets can be revoked as quickly and as easily as it was granted.
+
+Consequences:
+
+* We will need to create a vault, which takes time and costs money.
+* Some synchronization will be required on how to use the vault and what
+  secrets to store in it.
+* Secrets that are required by the easi-backend app will have to be updated in
+  SSM anyway, which means changing them requires updating the value in two
+  places (1Password and SSM).
+
+## Pros and Cons of the Alternatives
+
+### Email
+
+* `+` Fast.
+* `+` Some of our peers share secrets with us using email, so forwarding the
+  emails is an easy and straightforward way to distribute secrets.
+* `-` Poor access controls: once an email is sent, its contents cannot be recalled.
+* `-` Emails can be easily shared with a wide audience inadvertently, resulting
+  in a pervasive risk of uncontrolled disclosure.
+* `-` If we have to manage any secrets that cannot be rotated, there will be no
+  way to guarantee that sent emails are burned after reading: rather, they will
+  likely live on in the archived email of whoever received them.
+* `-` There would be no central source of truth on what the current secrets
+  are, which complicates onboarding/off-boarding, engineers returning from
+  vacation, etc.
+
+### Slack DMs
+
+* `+` It's easy to find people in Slack, and we all use it.
+* `+` Only two people are in a DM thread. Unlike email, it is not easy to share
+  contents of a DM inadvertently.
+* `-` Due to CMS record-keeping policies, DMs cannot be deleted and may be
+  subject to subpoena or FOIA requests, which makes them public records.
```

Comment:
```suggestion
* `-` Not a secure way to transfer such information.
```

Reply:
Tell me more about this - in my on-boarding training, the fact that slack DMs can be (under certain circumstances) subject to FOIA requests was specifically pointed out as a reason to avoid transmitting secrets that way.

Reply:
I'm not disagreeing. My initial version of this edit was: "Not a secure way to transfer such information", so you can feel free to edit the suggestion. But if you have a public link describing the policy, that would be helpful to include. I also thought the advice not to use Slack was applicable regardless of which Slack is used.

Reply:
What you're saying makes sense, but the suggested edit seems to lose all this detail. Specificity about _why_ DMs are insecure seems useful, especially from the POV of a future person reading this ADR who hasn't put much thought into it previously.

Reply:
791a144

Reply:
Looks good. Might be worth mentioning that Slack in general, regardless of the Slack workspace, shouldn't be used for this.

Reply:
> Might be worth mentioning that Slack in general, regardless of the Slack workspace, shouldn't be used for exchanging such information.

Yeah, this is important. Anything that's not 1Password or SSM is wrong. But I couldn't think of a good way to get this across in the ADR. But there will be some changes to make in the onboarding docs and whatnot to document how to use 1password on this project, and we can be sure to mention it there.